addons/purchase/purchase.py

```diff
@@ -786,8 +792,14 @@ def _get_purchase_order_date(self, cr, uid, procurement, company, schedule_date,
            :rtype: datetime
            :return: the desired Order Date for the PO
         """
-        seller_delay = int(procurement.product_id._select_seller(procurement.product_id).delay)
-        return schedule_date - relativedelta(days=seller_delay)
+
+        seller = procurement.product_id._select_seller(
+            procurement.product_id,
+            quantity=procurement.product_qty,
+            date=schedule_date.strftime(DEFAULT_SERVER_DATETIME_FORMAT)[:10],
```

Comment:
Suggestion: if you just want the date part, use `schedule_date.strftime(DEFAULT_SERVER_DATE_FORMAT)` instead of cutting part of the string.